src/SIL.Harmony/CrdtConfig.cs

```diff
@@ -8,14 +8,15 @@
 using SIL.Harmony.Entities;
 
 namespace SIL.Harmony;
-
+public delegate ValueTask BeforeSaveObjectDelegate(object obj, ObjectSnapshot snapshot);
 public class CrdtConfig
 {
     /// <summary>
     /// recommended to increase query performance, as getting objects can just query the table for that object.
     /// it does however increase database size as now objects are stored both in snapshots and in their projected tables
     /// </summary>
     public bool EnableProjectedTables { get; set; } = true;
+    public BeforeSaveObjectDelegate BeforePersistObject { get; set; } = (o, snapshot) => ValueTask.CompletedTask;
```

Comment:
Can we unify `BeforeSave` vs `BeforePersist`. I'd lean towards `Save`, because we're building on ef and it seems to have picked that term.

Reply:
oh haha, yeah that makes sense, I didn't realize I had changed that